src/linux/net/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -366,6 +450,19 @@ AccessPointControllerLinux::SetAkmSuites(std::vector<Ieee80211AkmSuite> akmSuite
         return status;
     }
 
+    // Validate that the AKM suites are successfully set by hostapd.
+    auto hostapdBssConfiguration = m_hostapd.GetConfiguration();
+    auto actualKeyManagements = hostapdBssConfiguration.WpaKeyMgmt;
+    if (std::equal(std::begin(wpaKeyManagements), std::end(wpaKeyManagements), std::begin(actualKeyManagements))) {
```

Comment:
(future reference only, no need to fix now)

nit: Any time iterators are being used in a read-only fashion (as here), the `const` iterator versions should be used instead:

```suggestion
    if (std::equal(std::cbegin(wpaKeyManagements), std::cend(wpaKeyManagements), std::cbegin(actualKeyManagements))) {
```